scripts/validate_docstrings.py

```diff
@@ -145,10 +146,23 @@ def examples_errors(self):
         runner = doctest.DocTestRunner(optionflags=flags)
         context = {"np": numpy, "pd": pandas}
         error_msgs = ""
+        current_dir = set(os.listdir())
         for test in finder.find(self.raw_doc, self.name, globs=context):
             f = io.StringIO()
             runner.run(test, out=f.write)
             error_msgs += f.getvalue()
+        leftovers = set(os.listdir()).difference(current_dir)
```

Comment:
if you do this why do we need skips?

Reply:
This is to validate that no doctest _ever_ creates leftover files. So the workflow is

1) Check if a docstring leftover a file/folder
2) If so, clean up those leftover files
3) Short-circut the validation and raise that a docstring leftover a file/folder

May helps potentially avoid any ResourceWarnings or potential segfaults from the docstring validation due to created files.

Reply:
sure i get it, but you added skips to all of the doc-strings hence they are not running?

Reply:
Correct.

IMO those docstrings that show an example with interacting with files should be skipped since they are more of a demonstration. The same behavior interacting with our file should be sufficient tested in the test suite.

Also I think it's better than having `os.remove` everywhere in the docstrings.